app/src/main/java/com/example/sopt_main/SignInActivity.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.sopt_main
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import com.example.sopt_main.databinding.ActivityMainBinding
+
+class SignInActivity : AppCompatActivity() {
+    private lateinit var binding : ActivityMainBinding
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        lateinit var resultLauncher: ActivityResultLauncher<Intent>
+        resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult())
+        { result ->
+            if (result.resultCode == Activity.RESULT_OK){
+                val id = result.data?.getStringExtra("id")
+                val password = result.data?.getStringExtra("pwd")
+                binding.idText.setText(id)
+                binding.pwdText.setText(password)
+            }
+        }
```

Comment:
지금 이코드를 보면 변수를 선언한 바로 아래에서 변수를 초기화하고 있는데, 그럴 경우에 그냥

```suggestion
        val resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult())
        { result ->
            if (result.resultCode == Activity.RESULT_OK){
                val id = result.data?.getStringExtra("id")
                val password = result.data?.getStringExtra("pwd")
                binding.idText.setText(id)
                binding.pwdText.setText(password)
            }
        }
```

`val`키워드를 사용해도 괜찮을 것 같으 :)

Reply:
아하 val로 바로초기화!! 확인했숩니당ㅎㅎ